tests/test_unit/test_sample_data.py

```diff
@@ -122,33 +141,140 @@ def test_fetch_metadata(tmp_path, caplog, download_fails, local_exists):
+    # Check frame_path
+    frame_path = getattr(ds, "frame_path", None)
+    if sample_metadata["frame_file"]:
+        assert frame_path is not None
+        assert frame_path.split("/")[-1] == sample_metadata["frame_file"]
+    else:
+        assert frame_path is None
 
-        if sample["video_file"] and with_video:
-            assert video_path.split("/")[-1] == sample["video_file"]
-        else:
-            assert video_path is None
+    # Check video_path
+    video_path = getattr(ds, "video_path", None)
+    if sample_metadata["video_file"] and with_video:
+        assert video_path is not None
+        assert video_path.split("/")[-1] == sample_metadata["video_file"]
+    else:
+        assert video_path is None
+
+
+@pytest.mark.parametrize(
+    "sample_name, expected_exception",
+    [
+        ("nonexistent_file", ValueError),
+        ("TRex_five-locusts.zip", NotImplementedError),
+    ],
+    ids=["invalid_file", "TRex_folder_zip"],
+)
+def test_fetch_dataset_invalid(sample_name, expected_exception):
+    with pytest.raises(expected_exception):
+        fetch_dataset(sample_name)
+
+
+def test_fetch_dataset_paths(sample_dataset):
+    """Test that the returned pose paths points to correct location.
+
+    If the pose files are in a zipped folder, the path should point to the
+    unzipped folder, otherwise it should point to the file itself.
+    """
+    sample_name, _ = sample_dataset
+    paths = fetch_dataset_paths(sample_name)
+    data_path = Path(paths.get("poses", paths.get("bboxes")))
+
+    if sample_name.endswith(".zip"):
+        # If the sample is a zip file,
+        # the path should point to the unzipped folder
+        assert data_path.is_dir()
+        assert data_path.name == sample_name.replace(".zip", "")
+        file_paths = list(data_path.iterdir())
+        assert len(file_paths) > 1
+        # Make sure unwanted files are not present
+        assert not any(
+            file_path.name in {".DS_Store", "Thumbs.db", "desktop.ini"}
+            for file_path in file_paths
+        )
+    else:
+        # If the sample is a single file,
+        # the path should point to the file itself
+        assert data_path.is_file()
+        assert data_path.name == sample_name
+
+
+@pytest.mark.parametrize(
+    "failing_func,expected_warning",
+    [
+        pytest.param(
+            "copy2",
+            "Failed to copy file",
+            id="copy2_exception",
+        ),
+        pytest.param(
+            "rmtree",
+            "Failed to remove unzip dir",
+            id="rmtree_exception",
+        ),
+    ],
+)
+def test_fetch_and_unzip_exceptions(
+    tmp_path, monkeypatch, caplog, failing_func, expected_warning
+):
+    """Test _fetch_and_unzip handles failures to copy files
+    or remove dirs gracefully, raising the appropriate warning.
+
+    It also tests that unwanted files are not copied to the new dest folder.
+    """
+    from movement import sample_data
```

Comment:
we already import from sample_data at the top, might as well import SAMPLE_DATA and _fetch_and_unzip there